swim/node.go

```diff
@@ -384,51 +375,22 @@ func (n *Node) Bootstrap(opts *BootstrapOptions) ([]string, error) {
 	return joined, nil
 }
 
-func (n *Node) seedBootstrapHosts(opts *BootstrapOptions) (err error) {
-	var hostports []string
-
-	n.bootstrapHosts = make(map[string][]string)
-
-	if opts.Hosts != nil {
-		hostports = opts.Hosts
-	} else {
-		switch true {
-		case true:
-			if opts.File != "" {
-				hostports, err = util.ReadHostsFile(opts.File)
-				if err == nil {
-					break
-				}
-				n.log.WithField("file", opts.File).Warnf("could not read host file: %v", err)
-			}
-			fallthrough
-		case true:
-			if n.bootstrapFile != "" {
-				hostports, err = util.ReadHostsFile(n.bootstrapFile)
-				if err == nil {
-					break
-				}
-				n.log.WithField("file", n.bootstrapFile).Warnf("could not read host file: %v", err)
-			}
-			fallthrough
-		case true:
-			hostports, err = util.ReadHostsFile("./hosts.json")
-			if err == nil {
-				break
-			}
-			n.log.WithField("file", "./hosts.json").Warnf("could not read host file: %v", err)
-			return errors.New("unable to read hosts file")
-		}
+// resolveDiscoverProvider reads BootstrapOptions and determines which
+// DiscoverProvider implementation to use for enumerating a list of bootstrap
+// hosts.
+// TODO: Remove this and legacy File/Hosts fields from BootstrapOptions and
+// accept only a DiscoverProvider.
+func (n *Node) resolveDiscoverProvider(opts *BootstrapOptions) (DiscoverProvider, error) {
```

Comment:
n isn't used, right? Does it make sense to separate the DiscoverProvider from the swin.Node? This function can then be moved to node_bootstrap.go. Let me know what you think.